lib/token.h

```diff
@@ -187,6 +187,30 @@ class CPPCHECKLIB Token {
     explicit Token(TokensFrontBack *tokensFrontBack = nullptr);
     ~Token();
 
+    struct TokenEnumeration
+    {
+        Token const* mt;
```

Comment:
It should probably store the start and the end, so it can iterate over partial ranges.

Reply:
you're right. I have a local change that does that which I'll push some time soon.